eno-core/src/main/java/fr/insee/eno/core/model/question/table/TableCell.java

```diff
@@ -0,0 +1,57 @@
+package fr.insee.eno.core.model.question.table;
+
+import datacollection33.GridResponseDomainInMixedType;
+import datacollection33.SelectDimensionType;
+import fr.insee.eno.core.annotations.DDI;
+import fr.insee.eno.core.annotations.Lunatic;
+import fr.insee.eno.core.model.EnoIdentifiableObject;
+import fr.insee.eno.core.model.response.Response;
+import fr.insee.eno.core.parameter.Format;
+import fr.insee.lunatic.model.flat.BodyCell;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.math.BigInteger;
+import java.util.List;
+
+import static fr.insee.eno.core.annotations.Contexts.Context;
+
+/** A TableCell object is the content of a table.
+ * A cell is neither part of the header nor of the left column.
+ * @see fr.insee.eno.core.model.question.TableQuestion */
+@Getter
+@Setter
+@Context(format = Format.DDI, type = GridResponseDomainInMixedType.class)
+@Context(format = Format.LUNATIC, type = BodyCell.class)
+public abstract class TableCell extends EnoIdentifiableObject {
+
+    /** Source parameter id from DDI **/
+    @DDI("getResponseDomain().getOutParameter().getIDArray(0).getStringValue()")
+    @Lunatic("setId(#param)")
+    String id;
+
+    /** Row position in the table. Starts at 1. */
+    @DDI("T(fr.insee.eno.core.model.question.table.TableCell).convertDDIDimension(#this, 1)")
+    Integer rowNumber;
+
+    /** Column position in the table. Starts at 1. */
+    @DDI("T(fr.insee.eno.core.model.question.table.TableCell).convertDDIDimension(#this, 2)")
+    Integer columnNumber;
+
+    /** Response object for Lunatic.
+     * In DDI, response names are mapped in the table question object and inserted here through a processing. */
+    @Lunatic("setResponse(#param)")
+    Response response;
+
+    public static Integer convertDDIDimension(GridResponseDomainInMixedType gridType, long dimensionNumber) {
+        List<SelectDimensionType> dimensions = gridType
+                .getGridAttachmentArray(0).getCellCoordinatesAsDefinedArray(0).getSelectDimensionList()
+                .stream()
+                .filter(selectDimensionType -> selectDimensionType.getRank().equals(BigInteger.valueOf(dimensionNumber)))
+                .toList();
+        if (dimensions.isEmpty())
+            return null;
```

Comment:
- add a bit of javadoc
- replace list with optional
- throw an exception instead of returning null if empty to avoid null pointer surprises

Reply:
We actually must return null if empty: in some cases (e.g. complex MCQ) DDI GridResponseDomainInMixed elements only have a rank '1' dimension.

Reply:
https://github.com/InseeFr/Eno/pull/633/commits/d997be53c6dc0463c6afb1567669d0656ab6ef0c